jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/java/org/jbpm/console/ng/cm/client/milestones/CaseMilestoneListSearchViewImpl.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.console.ng.cm.client.milestones;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.jboss.errai.common.client.dom.Button;
+import org.jboss.errai.common.client.dom.Form;
+import org.jboss.errai.common.client.dom.HTMLElement;
+import org.jboss.errai.common.client.dom.MouseEvent;
+import org.jboss.errai.databinding.client.api.DataBinder;
+import org.jboss.errai.databinding.client.api.StateSync;
+
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.jboss.errai.ui.shared.api.annotations.AutoBound;
+
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.ForEvent;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.jbpm.console.ng.cm.client.util.AbstractView;
+import org.jbpm.console.ng.cm.util.CaseMilestoneSearchRequest;
+
+
+import static org.jboss.errai.common.client.dom.DOMUtil.*;
+import static org.jbpm.console.ng.cm.client.resources.i18n.Constants.*;
+
+@Dependent
+@Templated("CaseMilestoneListViewImpl.html#search-actions")
+public class CaseMilestoneListSearchViewImpl extends AbstractView<CaseMilestoneListPresenter> {
+
+    @Inject
+    @DataField("search-actions")
+    private Form actions;
+
+    @Inject
+    @DataField("sort-alpha-asc")
+    private Button sortAphaAsc;
+
+    @Inject
+    @DataField("sort-alpha-desc")
+    private Button sortAphaDesc;
+
+    @Inject
+    @AutoBound
+    private DataBinder<CaseMilestoneSearchRequest> searchRequest;
+
+    @Override
+    public HTMLElement getElement() {
+        return actions;
+    }
+
+    @Inject
+    protected TranslationService translationService;
+
+    private CaseMilestoneListPresenter presenter;
+
+    @Override
+    public void init(final CaseMilestoneListPresenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @PostConstruct
+    public void init() {
+        searchRequest.setModel(new CaseMilestoneSearchRequest(), StateSync.FROM_MODEL);
+        searchRequest.addPropertyChangeHandler( e -> presenter.searchCaseMilestones() );
+        sortAphaAsc.setTitle(translationService.format(SORT_MILESTONES_BY_NAME));
```

Comment:
You can set these using the properties file instead, no need to use the translation service. Check this one: https://github.com/droolsjbpm/jbpm-console-ng/blob/master/jbpm-console-ng-case-mgmt/jbpm-console-ng-case-mgmt-client/src/main/resources/org/jbpm/console/ng/cm/client/resources/i18n/Constants.properties#L45